### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by jacobwindsor**
dcp-383, ebi-ait/dcp-ingest-central#383

- Upgrades all dependencies to latest (a lot of the changes are for API changes from this upgrade)
- Increases cache times for all static assets (JS, CSS) to a year
  - We can safely do this since we're using chunkhashes on builds so the cache will be busted on a release
- Removes cache for index.html
  - This will ensure latest chunkhashes are always used
- Sets the cache for all pages (non angular, if we at some point have them) to a day
  - Images and font files will be cached for a day but users will get updates quickly
- Sets staging builds to enable optimization and chunkhashes
  - So that the dev environment benefits from caching too and keeps our two environments closer together
- Adds docker instructions to the README
- Removes the prettier GH action until we find something better

### CODE REVIEW

File: e2e/tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "outDir": "../out-tsc/e2e",
     "module": "commonjs",
-    "target": "es5",
+    "target": "es2018",
```
Comment (ke4) [inline e2e/tsconfig.json:6]: In the tsconfig.json the value is `es2020`. Is there an equivalent here or this is the latest and matching?

**Review by ke4 — APPROVED** [review]
I just added a tiny question. I don't know the answer. 😏

File: e2e/tsconfig.json (hunk 1 above)
Comment (jacobwindsor) [reply e2e/tsconfig.json:6]: That is a good point. This was automatically updated by angular CLI so I assume it is correct. However, currently there's no e2e tests so I don't think we have to worry right now

Comment (ke4) [reply e2e/tsconfig.json:6]: OK, I already approved it, just thought I would ask it.

**Review by amnonkhen — APPROVED** [review]

**Review by aaclan-ebi — COMMENTED** [review]
noice! it's in Angular 12 now!

**Pull request merged**